`test_fix_line_ending_chinese_characters` fails on s390x

The `test_fix_line_ending_chinese_characters` test is dependent on system endianness because it doesn't define a byte-order mark (BOM) allowing Python methods like `replace()` to correctly determine endianness of UTF-16 characters. We therefore need to explicitly specify the UTF-16 endianness of the character when encoding it to bytes in the test, or it will fail like this on big-endian systems like s390x:
```
=================================== FAILURES ===================================
___________________ test_fix_line_ending_chinese_characters ____________________
    def test_fix_line_ending_chinese_characters() -> None:
        character = bytes('不', 'utf16')
        content = b''.join([character, b'\r\n', character, b'\n', character])
        expected = b''.join([character, b'\n', character, b'\n', character])
>       assert fix_line_ending(content) == expected
E       AssertionError: assert b'\xfe\xffN\r...\n\xfe\xffN\r' == b'\xfe\xffN\r...\n\xfe\xffN\r'
E         
E         At index 8 diff: b'\n' != b'\r'
E         
E         Full diff:
E         - (b'\xfe\xffN\r\n\xfe\xffN\r\n\xfe\xffN\r')
E         ?                           --
E         + (b'\xfe\xffN\r\n\xfe\xffN\n\xfe\xffN\r')
/builds/strophy/aports/testing/subliminal/src/subliminal-2.4.0/tests/test_subtitle.py:209: AssertionError
=========================== short test summary info ============================
FAILED tests/test_subtitle.py::test_fix_line_ending_chinese_characters - AssertionError: assert b'\xfe\xffN\r...\n\xfe\xffN\r' == b'\xfe\xffN\r...\n\xfe\xffN\r'
  
  At index 8 diff: b'\n' != b'\r'
  
  Full diff:
  - (b'\xfe\xffN\r\n\xfe\xffN\r\n\xfe\xffN\r')
  ?                           --
  + (b'\xfe\xffN\r\n\xfe\xffN\n\xfe\xffN\r')
=========== 1 failed, 596 passed, 16 skipped, 1 deselected in 51.08s ===========
```
The following patch fixes the test:
```
diff --git a/tests/test_subtitle.py b/tests/test_subtitle.py
index 217506d..c866d14 100644
--- a/tests/test_subtitle.py
+++ b/tests/test_subtitle.py
@@ -203,7 +203,7 @@ def test_fix_line_ending() -> None:
 
 # https://github.com/pannal/Sub-Zero.bundle/issues/646 replaced all Chinese character “不” with “上”
 def test_fix_line_ending_chinese_characters() -> None:
-    character = bytes('不', 'utf16')
+    character = bytes('不', 'utf-16-le')
     content = b''.join([character, b'\r\n', character, b'\n', character])
     expected = b''.join([character, b'\n', character, b'\n', character])
     assert fix_line_ending(content) == expected
```

--- a/changelog.d/1323.misc.rst
+++ b/changelog.d/1323.misc.rst
@@ -1,0 +1,1 @@
+fix test on big-endian system

--- a/tests/test_subtitle.py
+++ b/tests/test_subtitle.py
@@ -203,7 +203,7 @@ def test_fix_line_ending() -> None:
 
 # https://github.com/pannal/Sub-Zero.bundle/issues/646 replaced all Chinese character “不” with “上”
 def test_fix_line_ending_chinese_characters() -> None:
-    character = bytes('不', 'utf16')
+    character = bytes('不', 'utf-16-le')
     content = b''.join([character, b'\r\n', character, b'\n', character])
     expected = b''.join([character, b'\n', character, b'\n', character])
     assert fix_line_ending(content) == expected

